src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -193,4 +204,26 @@ private function addSingleProductVariantToOrder(ProductVariantInterface $product
 
         $this->objectManager->flush();
     }
+
+    /**
+     * @Given the customer bought single :product using :coupon coupon
+     */
+    public function theCustomerBoughtSingleUsing(ProductInterface $product, CouponInterface $coupon)
```

Comment:
Can't we extract some common logic to separate function. In OrderContext we have two other steps which are part of buying. I bet we can avoid some duplication there.